index.js

```diff
@@ -1,137 +1,145 @@
-let _fs
+let realFs
 try {
-  _fs = require('graceful-fs')
+  realFs = require('graceful-fs')
 } catch (_) {
-  _fs = require('fs')
+  realFs = require('fs')
 }
 const universalify = require('universalify')
+const memoize = require('memoize-weak')
```

Comment:
Why is this needed?

Reply:
The `jsonFileFactory` function is safe and idempotent. It doesn't do anything to the passed `fs` object, it just retains it in its closure. So it's safe to memoize for performance reasons, and the `memoize-weak` library does that without risking memory leaks.

Reply:
Have we actually run benchmarks here on the performance benefits? My guess is that there isn't enough computation being done on each call to warrant memoization.

Reply:
No benchmarks! It’s a habit of mine to memoize when I’m thinking about a performance-critical path, like the FS in a Webpack compiler, but I have no proof, and I can remove it if you want.

Reply:
Yeah, let's remove. Given that this is used by `fs-extra`, we try to keep deps light, given the number of things that use it.

Reply:
Done.